homeassistant/components/sonarr/sensor.py

```diff
@@ -72,198 +61,481 @@
+            key="queue",
+            name=f"{sonarr.app.info.app_name} Queue",
+            unit_of_measurement="Episodes",
+            enabled_default=False,
+        )
 
-    def update(self):
-        """Update the data for the sensor."""
+    @sonarr_exception_handler
+    async def async_update(self) -> None:
+        """Update entity."""
+        self._queue = await self.sonarr.queue()
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        attrs = {}
+
+        for item in self._queue:
+            remaining = 1 if item.size == 0 else item.size_remaining / item.size
+            remaining_pct = 100 * (1 - remaining)
+            name = f"{item.episode.series.title} {item.episode.identifier}"
+            attrs[name] = f"{remaining_pct:.2f}%"
+
+        return attrs
+
+    @property
+    def state(self) -> Union[None, str, int, float]:
+        """Return the state of the sensor."""
+        return len(self._queue)
+
+
+class SonarrSeriesSensor(SonarrSensor):
+    """Defines a Sonarr Series sensor."""
+
+    def __init__(self, sonarr: Sonarr, entry_id: str, unique_id: str) -> None:
+        """Initialize Sonarr Series sensor."""
+        self._items = []
+
+        super().__init__(
+            sonarr=sonarr,
+            entry_id=entry_id,
+            unique_id=unique_id,
+            icon="mdi:television",
+            key="series",
+            name=f"{sonarr.app.info.app_name} Shows",
+            unit_of_measurement="Series",
+            enabled_default=False,
+        )
+
+    @sonarr_exception_handler
+    async def async_update(self) -> None:
+        """Update entity."""
+        self._items = await self.sonarr.series()
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        attrs = {}
+
+        for item in self._items:
+            attrs[item.series.title] = f"{item.downloaded}/{item.episodes} Episodes"
+
+        return attrs
+
+    @property
+    def state(self) -> Union[None, str, int, float]:
+        """Return the state of the sensor."""
+        return len(self._items)
+
+
+class SonarrStatusSensor(SonarrSensor):
+    """Defines a Sonarr Status sensor."""
+
+    def __init__(self, sonarr: Sonarr, entry_id: str, unique_id: str) -> None:
+        """Initialize Sonarr Status sensor."""
+        self._version = None
+
+        super().__init__(
+            sonarr=sonarr,
+            entry_id=entry_id,
+            unique_id=unique_id,
+            icon="mdi:information",
+            key="status",
+            name=f"{sonarr.app.info.app_name} Status",
+            unit_of_measurement=None,
+            enabled_default=False,
+        )
+
+    @sonarr_exception_handler
+    async def async_update(self) -> None:
+        """Update entity."""
+        app = await self.sonarr.update()
+        self._version = app.info.version
```

Comment:
this seems like something you should return as the firmware version in device info ? Shouldn't be a sensor?

Reply:
i was going for parity with former sensors. I believe i do set in device info as well.

Reply:
im happy to remove as i thought it was a bit useless too